crates/ika-move-packages/packages/ika_system/sources/dwallet/dwallet_2pc_mpc_coordinator_inner.move

```diff
@@ -2250,6 +2253,7 @@ public(package) fun advance_epoch(
 ): Balance<IKA> {
     assert!(self.pricing_and_fee_management.calculation_votes.is_none(), EPricingCalculationVotesMustBeCompleted);
     assert!(self.all_current_epoch_sessions_completed(), ECannotAdvanceEpoch);
+    assert!(self.received_end_of_publish || self.current_epoch == 0, ECannotAdvanceEpoch);
```

Comment:
why shouldn't we receive an end of publish for the first epoch?

Reply:
```suggestion
    // We advance the first epoch `0` immediately during initialization, the network doesn't participate in it and therefore, it did not send an `END_OF_PUBLISH`. For any other epoch, don't advance before the network sent an `END_OF_PUBLISH`.
    assert!(self.received_end_of_publish || self.current_epoch == 0, ECannotAdvanceEpoch);
```

Reply:
create the object with received_end_of_publish = true so you don't have to check the first epoch